src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -1619,10 +1623,48 @@ internal static void ApplyConfiguration(JObject config, ScriptHostConfiguration
             }
             scriptConfig.HostConfig.FunctionTimeout = ScriptHost.CreateTimeoutConfiguration(scriptConfig);
 
+            ApplyLanguageWorkerConfig(config, scriptConfig, logger);
             ApplyLoggerConfig(config, scriptConfig);
             ApplyApplicationInsightsConfig(config, scriptConfig);
         }
 
+        private static void ApplyLanguageWorkerConfig(JObject config, ScriptHostConfiguration scriptConfig, ILogger logger)
+        {
+            JToken value = null;
+            JObject languageWorkerSection = (JObject)config["languageWorker"];
+            int requestedGrpcMaxMessageLength = DefaultMaxMessageLengthBytes;
+            if (languageWorkerSection != null)
+            {
+                if (languageWorkerSection.TryGetValue("maxMessageLength", out value))
+                {
+                    requestedGrpcMaxMessageLength = int.Parse((string)value);
+                    if (ScriptSettingsManager.Instance.IsDynamicSku)
+                    {
+                        string message = $"Cannot set {nameof(scriptConfig.MaxMessageLengthBytes)} on Consumption plan. Default MaxMessageLength: {DefaultMaxMessageLengthBytesDynamicSku} will be used";
+                        logger?.LogWarning(message);
+                        requestedGrpcMaxMessageLength = DefaultMaxMessageLengthBytesDynamicSku;
```

Comment:
as discussed, this line wont get hit if they have a language worker section but no max size, and thus the default won't be set to the dynamic default. Add a test case for this and you should see the failure